### ISSUE TITLE
Invalid escape character generated from valid regex pattern in swagger yaml

### ISSUE THREAD

**Issue by okthatsneat** [zalando/play-swagger#53]
attribute causing the issue: 
```yaml
      keywords:
        description: |
          Product Keywords. Comma-Space (', ') separated Keyword List. A keyword may contain letters
          both upper and lowercase as well as numbers and spaces. Each keyword has a maximum length of 255 characters.
        type: string
        pattern: /([\w\s]{1,255}|([\w\s]{1,255}, )+[\w\s]{1,255})/

```

```bash
[info] Compiling 15 Scala sources and 1 Java source to /Users/phofmann/code/ArticleValidator2/target/scala-2.11/classes...
[error] /Users/phofmann/code/ArticleValidator2/target/scala-2.11/routes/main/validators/echo.yaml.scala:10: invalid escape character
[error]         Seq(pattern("/([\w\s]{1,255}|([\w\s]{1,255}, )+[\w\s]{1,255})/".r))
[error]                          ^
[error] /Users/phofmann/code/ArticleValidator2/target/scala-2.11/routes/main/validators/echo.yaml.scala:10: invalid escape character
[error]         Seq(pattern("/([\w\s]{1,255}|([\w\s]{1,255}, )+[\w\s]{1,255})/".r))
[error]                            ^
[error] /Users/phofmann/code/ArticleValidator2/target/scala-2.11/routes/main/validators/echo.yaml.scala:10: invalid escape character
[error]         Seq(pattern("/([\w\s]{1,255}|([\w\s]{1,255}, )+[\w\s]{1,255})/".r))
[error]                                         ^
[error] /Users/phofmann/code/ArticleValidator2/target/scala-2.11/routes/main/validators/echo.yaml.scala:10: invalid escape character
[error]         Seq(pattern("/([\w\s]{1,255}|([\w\s]{1,255}, )+[\w\s]{1,255})/".r))
[error]                                           ^
[error] /Users/phofmann/code/ArticleValidator2/target/scala-2.11/routes/main/validators/echo.yaml.scala:10: invalid escape character
[error]         Seq(pattern("/([\w\s]{1,255}|([\w\s]{1,255}, )+[\w\s]{1,255})/".r))
[error]                                                          ^
[error] /Users/phofmann/code/ArticleValidator2/target/scala-2.11/routes/main/validators/echo.yaml.scala:10: invalid escape character
[error]         Seq(pattern("/([\w\s]{1,255}|([\w\s]{1,255}, )+[\w\s]{1,255})/".r))
[error]                                                            ^
[error] 6 errors found
[error] (compile:compileIncremental) Compilation failed
[error] application -

! @6oej351kn - Internal server error, for (GET) [/] ->

play.sbt.PlayExceptions$CompilationException: Compilation error[invalid escape character]
	at play.sbt.PlayExceptions$CompilationException$.apply(PlayExceptions.scala:27) ~[na:na]
	at play.sbt.PlayExceptions$CompilationException$.apply(PlayExceptions.scala:27) ~[na:na]
	at scala.Option.map(Option.scala:145) ~[scala-library-2.11.7.jar:na]
	at play.sbt.run.PlayReload$$anonfun$taskFailureHandler$1.apply(PlayReload.scala:49) ~[na:na]
	at play.sbt.run.PlayReload$$anonfun$taskFailureHandler$1.apply(PlayReload.scala:44) ~[na:na]
	at scala.Option.map(Option.scala:145) ~[scala-library-2.11.7.jar:na]
	at play.sbt.run.PlayReload$.taskFailureHandler(PlayReload.scala:44) ~[na:na]
	at play.sbt.run.PlayReload$.compileFailure(PlayReload.scala:40) ~[na:na]
	at play.sbt.run.PlayReload$$anonfun$compile$1.apply(PlayReload.scala:17) ~[na:na]
	at play.sbt.run.PlayReload$$anonfun$compile$1.apply(PlayReload.scala:17) ~[na:na]
```

**Comment by slavaschmidt** [comment]
In Java, you need to double escape backslashes, see for example http://www.ocpsoft.org/opensource/guide-to-regular-expressions-in-java-part-1/, 1.1
At the moment yaml file get's parsed, the double slashes are "unescaped" by jackson JSON parser. Thus, you actually need to "double double" escape backslashes. Here is an example of valid regex in swagger yaml file: 
``` "^([a-zA-Z0-9]+\\\\s?)*$" ```
We discussed a possibility to deal with this issue, but this behaviour is well know for experienced Java developers, so we decided to leave the default behaviour for now.

Not closing the issue because we might want to reconsider.

**Comment by okthatsneat** [comment]
thank you! i guess this needs to be documented as it's likely that people will use the editor.swagger.io to validate their swagger, which does not complain about the escaping. maybe the additional rules imposed on swagger formats should be validated with a tool from this project?

**Comment by slavaschmidt** [comment]
We should put information about proper escaping of java regexps in swagger specification into the documentation.

**Comment by slavaschmidt** [comment]
Fixed in 0.1.7
